agent.py

```diff
@@ -445,8 +445,13 @@ def _submit_jmx_service_discovery(self, jmx_sd_configs):
             return
 
         if self.supervisor_proxy is not None:
-            jmx_state = self.supervisor_proxy.supervisor.getProcessInfo(JMX_SUPERVISOR_ENTRY)
-            log.debug("Current JMX check state: %s", jmx_state['statename'])
+            try:
+                jmx_state = self.supervisor_proxy.supervisor.getProcessInfo(JMX_SUPERVISOR_ENTRY)
+                log.debug("Current JMX check state: %s", jmx_state['statename'])
+                return
```

Comment:
We want the `return` inside the except block instead of here, otherwise we'll be skipping the rest of the method when everything is fine, and not the opposite.